### PR TITLE
DB model for projects

### PR DESCRIPTION
**Pull request by anissa-agahchen**
# Overview

## Links to jira tickets

- {https://quartech.atlassian.net/browse/BHBC-658}

## This PR contains the following changes

- {DB models for Biohub projects}

## This PR contains the following types of changes

- [x] New feature (change which adds functionality)
- [ ] Bug fix (change which fixes an issue)
- [ ] Misc cleanup / Refactoring / Documentation
- [ ] Version change

## Checklist

A list of items that are good to consider when making any changes.

_Note: this list is not exhaustive, and not all items are always applicable._

### General

- [x] I have performed a self-review of my own code
- [ ] My code follows the style guidelines of this project
- [ ] New and existing unit tests pass locally with my changes

### Code

- [ ] New files/classes/functions have appropriately descriptive names and comment blocks to describe their use/behaviour
- [x] I have avoided duplicating code when possible, moving re-usable pieces into functions
- [x] I have avoided hard-coding values where possible and moved any re-usable constants to a constants file
- [ ] My code is as flat as possible (avoids deeply nested if/else blocks, promise chains, etc)
- [ ] My code changes account for null/undefined values and handle errors appropriately
- [ ] My code uses types/interfaces to help describe values/parameters/etc and inform future developers

### Documentation
- [ ] I have commented my code sufficiently, such that an unfamiliar developer could understand my code
- [ ] I have added/updated README's and related documentation

### Tests
- [ ] I have added/updated unit tests for any new code I've added/updated
- [ ] I have added/updated Postman tests to account for any new/updated code

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Screenshots

Please add any relevant UI screenshots, if applicable.

### CODE REVIEW

File: database/src/migrations/20210107132219_setup_biohub_db.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
+import Knex from 'knex';
+import * as fs from 'fs';
+
+const DB_SCHEMA = 'biohub';
+const path = require('path');
+
+export async function up(knex: Knex): Promise<void> {
+    //const filePath = path.join(__dirname, '..', 'release.0.3', 'biohub.0.3.sql')
```
Comment (NickPhura) [inline database/src/migrations/20210107132219_setup_biohub_db.ts]: old commented code

**Review by NickPhura — DISMISSED** [review]

**Comment by rstens** [comment]
In the database setup we are getting an error:

```
    grant usage on schema biohub_dapi_v1 to biohub_api;
    grant usage on schema  biohub to biohub_api;
    grant all on all tables in schema  biohub_dapi_v1  to biohub_api;
    alter DEFAULT PRIVILEGES in SCHEMA biohub_dapi_v1 grant ALL on tables to biohub_api;
    alter role biohub_api set search_path to biohub_dapi_v1, biohub;
   - permission denied to create role
    at Parser.parseErrorMessage (/opt/app-root/src/node_modules/pg-protocol/src/parser.ts:357:11)
    at Parser.handlePacket (/opt/app-root/src/node_modules/pg-protocol/src/parser.ts:186:21)
    at Parser.parse (/opt/app-root/src/node_modules/pg-protocol/src/parser.ts:101:30)
    at Socket.stream.on (/opt/app-root/src/node_modules/pg-protocol/src/index.ts:7:48)
    at Socket.emit (events.js:182:13)
    at addChunk (_stream_readable.js:283:12)
    at readableAddChunk (_stream_readable.js:264:11)
    at Socket.Readable.push (_stream_readable.js:219:10)
    at TCP.onStreamRead [as onread] (internal/stream_base_commons.js:94:17)
```

**Review by rstens — CHANGES_REQUESTED** [review]
We have a generic user defined who is used for accessing the database this is `biohubbc-user`.
This main user does not have to be created separately.
Also in the scrips this user does not get referred to by its parameter but by the actual hardcoded name.
Similarly the schema: deployment of the database will always create the "biohub" schema, so it does not have to be created again. But the current code is actually no problem as it check for existence.

Also, I reported an issue in another comment

File: database/src/migrations/20210107132219_setup_biohub_db.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
+import Knex from 'knex';
+import * as fs from 'fs';
+
+const DB_SCHEMA1 = 'biohub';
+const DB_SCHEMA2 = 'biohub_dapi_v1';
+const DB_USER = 'biohub_api';
+const path = require('path');
```
Comment (NickPhura) [inline database/src/migrations/20210107132219_setup_biohub_db.ts]: should avoid using require in typescript projects:

`import path from 'path';`

Not 100% sure on the ramifications, if any, but import is the preferred way

**Review by rstens — APPROVED** [review]

**Pull request merged**